pkg/slo-controller/noderesource/framework/extender_plugin.go

```diff
@@ -24,8 +24,17 @@ import (
 	"github.com/koordinator-sh/koordinator/pkg/slo-controller/metrics"
 )
 
+// node resource initialization:
+//  1. Setup
+//     node resource reconcile loop:
+//  1. Calculate / Reset (determined by whether the config disables node resources, e.g. colocation disabled)
+//  2. NodePreUpdate
+//  3. NodePrepare: (it should be idempotent since it can be called multiple times in one loop)
+//  4. NodeSync
+//  5. NodeMetaSync
 var (
 	globalSetupExtender             = NewRegistry("Setup")
+	globalNodePreUpdateExtender     = NewRegistry("NodePreUpdate")
 	globalNodePrepareExtender       = NewRegistry("NodePrepare")
```

Comment:
better to rename NodePreparePlugin.Execute to NodePreparePlugin.Prepare

Reply:
done